apps/browser/src/autofill/services/inline-menu-field-qualification.service.ts

```diff
@@ -0,0 +1,395 @@
+      if (visiblePasswordFieldsInPageDetails.length > 1) {
+        return false;
+      }
+
+      // If the page does not contain any password fields, it might be part of a multistep login form.
+      // That will only be the case if the field does not explicitly have its autocomplete attribute
+      // set to "off" or "false".
+      return !this.autocompleteDisabledValues.has(field.autoCompleteType);
+    }
+
+    // If the field is structured within a form, but no password fields are present in the form,
+    // we need to consider whether the field is part of a multistep login form.
+    if (passwordFieldsInPageDetails.length === 0) {
+      // If the field's autocomplete is set to a disabled value, we should assume that the field is
+      // not part of a login form.
+      if (this.autocompleteDisabledValues.has(field.autoCompleteType)) {
+        return false;
+      }
+
+      // If the form that contains the field has more than one visible field, we should assume
+      // that the field is part of an account creation form.
+      const fieldsWithinForm = pageDetails.fields.filter(
+        (pageDetailsField) => pageDetailsField.form === field.form && pageDetailsField.viewable,
+      );
+      return fieldsWithinForm.length === 1;
+    }
+
+    // If a single password field exists within the page details, and that password field is part of
+    // the same form as the provided field, we should assume that the field is part of a login form.
+    // If multiple visible password fields exist within the page details, we need to assume that the
+    // provided field is part of an account creation form.
+    const visiblePasswordFieldsInPageDetails = passwordFieldsInPageDetails.filter(
+      (passwordField) => passwordField.form === field.form && passwordField.viewable,
+    );
+    return visiblePasswordFieldsInPageDetails.length === 1;
+  }
+
+  /**
+   * Validates the provided field as a username field.
+   *
+   * @param field - The field to validate
+   */
+  private isUsernameField = (field: AutofillField): boolean => {
+    if (
+      !this.usernameFieldTypes.has(field.type) ||
+      this.isExcludedFieldType(field, this.excludedAutofillLoginTypesSet)
+    ) {
+      return false;
+    }
+
+    return this.keywordsFoundInFieldData(field, AutoFillConstants.UsernameFieldNames);
+  };
+
+  /**
+   * Validates the provided field as a current password field.
+   *
+   * @param field - The field to validate
+   */
+  private isCurrentPasswordField = (field: AutofillField): boolean => {
+    if (field.autoCompleteType === "new-password") {
+      return false;
+    }
+
+    return this.isPasswordField(field);
+  };
+
+  /**
+   * Validates the provided field as a new password field.
+   *
+   * @param field - The field to validate
+   */
+  private isNewPasswordField = (field: AutofillField): boolean => {
+    if (field.autoCompleteType === "current-password") {
+      return false;
+    }
+
```

Comment:
We could also shortcut this if the autocomplete type is explicitly a new password:
```suggestion

    if (field.autoCompleteType === "new-password") {
      return true;
    }

```

Reply:
> Hmm... so this pulls us closer to "trusting" the intent of a developer's markup... would that be desired?

I see the values of "new-password" and "current-password" more as descriptors than prescribing behaviour, whereas "off" is not helpful as a descriptor and arguably not trustworthy as a prescriber of behaviour to carry out user-intent.